client/src/pages/Home/Home.jsx

```diff
@@ -1,13 +1,41 @@
+import { useEffect, useState, useContext } from "react";
 import "../../../node_modules/bootstrap/dist/css/bootstrap.min.css";
 import CardWrapper from "../../components/ui/CardWrapper/CardWrapper";
-// here will get kinds from recommendation system
-const activity_types = ["beaches", "urban_environment", "architecture"];
+import { AuthContext } from "../../contexts/authContexts/authContexts";
+
 const Home = () => {
+  const [activityTypes, setActivityTypes] = useState([]);
+  const { currentUser } = useContext(AuthContext);
+
+  useEffect(() => {
+    if (currentUser && currentUser.uid) {
```

Comment:
there should be a centralized way of handling if a user is signed in. I think its called protected routes or something, where you can specify the routes of your app that can be accessed only after signing in, and else it will redirect to the sign in page.

Reply:
will come back to this- added to doc for later